tests/unit/test_clidriver.py

```diff
@@ -771,7 +771,7 @@ def test_user_agent(self, *args):
         machine = platform.machine()
         driver = create_clidriver()
         user_agent_extra_pattern = re.compile(
-            '^source/%s(\.[a-z_]+)?(\.[0-9]+)?$' % machine
+            r'^source/%s(\.[a-z_]+)?(\.[0-9]+)?$' % machine
```

Comment:
While we're here we might as well do:
```suggestion
            fr'^source/{machine}(\.[a-z_]+)?(\.[0-9]+)?$'
```